nrf_security/src/psa/Kconfig

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig PSA_NATIVE_ITS
+	bool "PSA native Internal Trusted Storage"
+	depends on CC310_BACKEND
+
+if PSA_NATIVE_ITS
+
+choice
+	prompt "Backend selection for PSA native ITS"
+	help
+	  Select backend for PSA native Internal Trusted Storage
+
+config CHOICE_BACKEND_ZEPHYR_SETTINGS
+	bool "Zephyr Settings"
+	select PSA_EITS_BACKEND_ZEPHYR
+
+endchoice
+
+menuconfig PSA_EITS_BACKEND_ZEPHYR
+	bool "Native ITS backend based on Zephyr Settings subsystem"
+	select FLASH
```

Comment:
@lmaciejonczyk  this is causing cyclic Kconfig dependencies.
I disabled it here: https://github.com/nrfconnect/sdk-nrfxlib/pull/426/commits/28e34ee7f79a3cc66b2ff10daa7b4ff12df61f6f

Could you take a second look at what is you real need and intentions, feel free to ping me directly:
```
ERROR   : Test KconfigBasic failed: Dependency loop
===============

GPIO (defined at boards/shields/adafruit_2_8_tft_touch_v2/Kconfig.defconfig:6, boards/arm/arty/Kconfig.defconfig:28, boards/shields/adafruit_2_8_tft_touch_v2/Kconfig.defconfig:6, soc/arm/microchip_mec/mec1501/Kconfig.defconfig.mec1501hsz:19, soc/arm/nordic_nrf/Kconfig.defconfig:41, soc/arm/nxp_imx/mcimx7_m4/Kconfig.defconfig.mcimx7_m4:14, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1010:27, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1015:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1021:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1024:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1052:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1062:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1064:23, soc/arm/nxp_kinetis/k2x/Kconfig.defconfig.mk22f12:33, soc/arm/nxp_kinetis/k6x/Kconfig.defconfig.mk64f12:37, soc/arm/nxp_kinetis/k6x/Kconfig.defconfig.mk66f18:28, soc/arm/nxp_kinetis/k8x/Kconfig.defconfig.series:44, soc/arm/nxp_kinetis/ke1xf/Kconfig.defconfig.series:61, soc/arm/nxp_kinetis/kv5x/Kconfig.defconfig.series:27, soc/arm/nxp_lpc/lpc54xxx/Kconfig.defconfig.lpc54114_m0:15, soc/arm/nxp_lpc/lpc55xxx/Kconfig.defconfig.lpc55S69_cpu1:15, drivers/gpio/Kconfig:6), with definition...

config GPIO
        bool
        default y
        depends on SHIELD_ADAFRUIT_2_8_TFT_TOUCH_V2
...
menuconfig GPIO
        bool "GPIO Drivers"
        select HAS_DTS_GPIO
        help
          Include GPIO drivers in system config

(select-related dependencies: DK_LIBRARY || NRF_SW_LPUART || (BOARD_NRF9160DK_NRF52840 && SOC_NRF52840_QIAA && <choice>) || (BOARD_THINGY52_NRF52832 && SOC_NRF52832_QFAA && <choice>) || (IEEE802154_RF2XX && NETWORKING && IEEE802154) || (WIFI_ESWIFI && WIFI))

(imply-related dependencies: DK_LIBRARY || NRF_SW_LPUART || (BOARD_NRF9160DK_NRF52840 && SOC_NRF52840_QIAA && <choice>) || (BOARD_THINGY52_NRF52832 && SOC_NRF52832_QFAA && <choice>) || (IEEE802154_RF2XX && NETWORKING && IEEE802154) || (WIFI_ESWIFI && WIFI))

...depends on SPI (defined at boards/shields/adafruit_2_8_tft_touch_v2/Kconfig.defconfig:11, boards/shields/adafruit_winc1500/Kconfig.defconfig:8, boards/shields/buydisplay_2_8_tft_touch_arduino/Kconfig.defconfig:8, boards/shields/buydisplay_3_5_tft_touch_arduino/Kconfig.defconfig:8, boards/shields/dac80508_evm/Kconfig.defconfig:8, boards/shields/dfrobot_can_bus_v2_0/Kconfig.defconfig:8, boards/shields/frdm_cr20a/Kconfig.defconfig:11, boards/shields/link_board_eth/Kconfig.defconfig:8, boards/shields/lmp90100_evb/Kconfig.defconfig:8, boards/shields/ls0xx_generic/Kconfig.defconfig:8, boards/shields/mikroe_adc_click/Kconfig.defconfig:8, boards/shields/mikroe_eth_click/Kconfig.defconfig:9, boards/shields/semtech_sx1272mb2das/Kconfig.defconfig:8, boards/shields/ssd1306/Kconfig.defconfig:10, boards/shields/st7735r/Kconfig.defconfig:8, boards/shields/st7789v_generic/Kconfig.defconfig:8, boards/shields/v2c_daplink/Kconfig.defconfig:10, boards/shields/waveshare_epaper/Kconfig.defconfig:13, boards/shields/x_nucleo_idb05a1/Kconfig.defconfig:8, boards/arm/96b_carbon/Kconfig.defconfig:17, boards/arm/arty/Kconfig.defconfig:47, boards/arm/b_l4s5i_iot01a/Kconfig.defconfig:30, boards/arm/b_l4s5i_iot01a/Kconfig.defconfig:52, boards/arm/decawave_dwm1001_dev/Kconfig.defconfig:14, boards/arm/disco_l475_iot1/Kconfig.defconfig:35, boards/arm/disco_l475_iot1/Kconfig.defconfig:57, boards/arm/frdm_k82f/Kconfig.defconfig:25, boards/arm/reel_board/Kconfig.defconfig:34, boards/arm/ruuvi_ruuvitag/Kconfig.defconfig:11, boards/arm/stm32l562e_dk/Kconfig.defconfig:18, boards/arm/thingy53_nrf5340/Kconfig.defconfig:80, boards/shields/adafruit_2_8_tft_touch_v2/Kconfig.defconfig:11, boards/shields/adafruit_winc1500/Kconfig.defconfig:8, boards/shields/buydisplay_2_8_tft_touch_arduino/Kconfig.defconfig:8, boards/shields/buydisplay_3_5_tft_touch_arduino/Kconfig.defconfig:8, boards/shields/dac80508_evm/Kconfig.defconfig:8, boards/shields/dfrobot_can_bus_v2_0/Kconfig.defconfig:8, boards/shields/frdm_cr20a/Kconfig.defconfig:11, boards/shields/link_board_eth/Kconfig.defconfig:8, boards/shields/lmp90100_evb/Kconfig.defconfig:8, boards/shields/ls0xx_generic/Kconfig.defconfig:8, boards/shields/mikroe_adc_click/Kconfig.defconfig:8, boards/shields/mikroe_eth_click/Kconfig.defconfig:9, boards/shields/semtech_sx1272mb2das/Kconfig.defconfig:8, boards/shields/ssd1306/Kconfig.defconfig:10, boards/shields/st7735r/Kconfig.defconfig:8, boards/shields/st7789v_generic/Kconfig.defconfig:8, boards/shields/v2c_daplink/Kconfig.defconfig:10, boards/shields/waveshare_epaper/Kconfig.defconfig:13, boards/shields/x_nucleo_idb05a1/Kconfig.defconfig:8, boards/xtensa/intel_s1000_crb/Kconfig.defconfig:137, soc/arm/nxp_kinetis/k6x/Kconfig.defconfig.mk64f12:52, soc/arm/nxp_kinetis/kwx/Kconfig.defconfig.mkw2xd512:45, drivers/spi/Kconfig:9), with definition...

config SPI
        bool
        default y
        depends on DISPLAY && SHIELD_ADAFRUIT_2_8_TFT_TOUCH_V2
...
menuconfig SPI
        bool "SPI hardware bus support"
        help
          Enable support for the SPI hardware bus.

(select-related dependencies: (IEEE802154_RF2XX && NETWORKING && IEEE802154) || (WIFI_ESWIFI_BUS_SPI && <choice WIFI_ESWIFI_BUS>) || (SDMMC_OVER_SPI && DISK_DRIVER_SDMMC && DISK_DRIVERS))

...depends on FLASH (defined at drivers/flash/Kconfig:35), with definition...

menuconfig FLASH
        bool "Flash hardware support"
        help
          Enable support for the flash hardware.

(select-related dependencies: (ITE_IT8XXX2_INTC && SOC_SERIES_RISCV32_IT8XXX2) || (PSA_EITS_BACKEND_ZEPHYR && CHOICE_BACKEND_ZEPHYR_SETTINGS && PSA_NATIVE_ITS && MBEDTLS_PSA_CRYPTO_C && NRF_SECURITY_ANY_BACKEND && NORDIC_SECURITY_BACKEND) || (EEPROM_EMULATOR && EEPROM) || (DISK_DRIVER_FLASH && DISK_DRIVERS) || (MCUMGR_CMD_IMG_MGMT && MCUMGR) || HAWKBIT || UPDATEHUB || (NET_L2_OPENTHREAD && NETWORKING && NETWORKING))

(imply-related dependencies: (ITE_IT8XXX2_INTC && SOC_SERIES_RISCV32_IT8XXX2) || (PSA_EITS_BACKEND_ZEPHYR && CHOICE_BACKEND_ZEPHYR_SETTINGS && PSA_NATIVE_ITS && MBEDTLS_PSA_CRYPTO_C && NRF_SECURITY_ANY_BACKEND && NORDIC_SECURITY_BACKEND) || (EEPROM_EMULATOR && EEPROM) || (DISK_DRIVER_FLASH && DISK_DRIVERS) || (MCUMGR_CMD_IMG_MGMT && MCUMGR) || HAWKBIT || UPDATEHUB || (NET_L2_OPENTHREAD && NETWORKING && NETWORKING))

...depends on NRF_SECURITY_ANY_BACKEND (defined at /projects/github/ncs/nrfxlib/nrf_security/Kconfig:181), with definition...

config NRF_SECURITY_ANY_BACKEND
        bool
        default y
        depends on (CC3XX_BACKEND || OBERON_BACKEND || MBEDTLS_VANILLA_BACKEND) && NORDIC_SECURITY_BACKEND

...depends on OBERON_BACKEND (defined at /projects/github/ncs/nrfxlib/nrf_security/Kconfig:155), with definition...
...
menuconfig CLOCK_CONTROL
        bool "Hardware clock controller support"
        help
          Enable support for hardware clock controller. Such hardware can
          provide clock for other subsystem, and thus can be also used for
          power efficiency by controlling their clock. Note that this has
          nothing to do with RTC.

(select-related dependencies: (BOARD_NRF52_BSIM && <choice>) || (SOC_SERIES_IMX_6X_M4 && <choice>) || (SOC_SERIES_IMX7_M4 && <choice>) || (SOC_SERIES_IMX_RT && <choice>) || (SOC_SERIES_IMX_RT6XX && <choice>) || (SOC_SERIES_KINETIS_K2X && <choice>) || (SOC_SERIES_KINETIS_K6X && <choice>) || (SOC_SERIES_KINETIS_K8X && <choice>) || (SOC_SERIES_KINETIS_KE1XF && <choice>) || (SOC_SERIES_KINETIS_KL2X && <choice>) || (SOC_SERIES_KINETIS_KV5X && <choice>) || (SOC_SERIES_KINETIS_KWX && <choice>) || (SOC_SERIES_LPC11U6X && <choice>) || (SOC_OPENISA_RV32M1_RISCV32 && <choice>) || (SOC_ESP32 && <choice>) || (SOC_LPC54114_M4 && <choice>) || (SOC_LPC54114_M0 && <choice>) || (SOC_LPC55S16 && <choice>) || (SOC_LPC55S28 && <choice>) || (SOC_LPC55S69_CPU0 && <choice>) || (GPIO_DW_CLOCK_GATE && GPIO_DW && GPIO))

...depends again on GPIO (defined at boards/shields/adafruit_2_8_tft_touch_v2/Kconfig.defconfig:6, boards/arm/arty/Kconfig.defconfig:28, boards/shields/adafruit_2_8_tft_touch_v2/Kconfig.defconfig:6, soc/arm/microchip_mec/mec1501/Kconfig.defconfig.mec1501hsz:19, soc/arm/nordic_nrf/Kconfig.defconfig:41, soc/arm/nxp_imx/mcimx7_m4/Kconfig.defconfig.mcimx7_m4:14, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1010:27, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1015:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1021:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1024:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1052:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1062:23, soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1064:23, soc/arm/nxp_kinetis/k2x/Kconfig.defconfig.mk22f12:33, soc/arm/nxp_kinetis/k6x/Kconfig.defconfig.mk64f12:37, soc/arm/nxp_kinetis/k6x/Kconfig.defconfig.mk66f18:28, soc/arm/nxp_kinetis/k8x/Kconfig.defconfig.series:44, soc/arm/nxp_kinetis/ke1xf/Kconfig.defconfig.series:61, soc/arm/nxp_kinetis/kv5x/Kconfig.defconfig.series:27, soc/arm/nxp_lpc/lpc54xxx/Kconfig.defconfig.lpc54114_m0:15, soc/arm/nxp_lpc/lpc55xxx/Kconfig.defconfig.lpc55S69_cpu1:15, drivers/gpio/Kconfig:6)
```

Reply:
> @lmaciejonczyk why is `nrf_security/include/native_its/psa/error.h` empty ?
> It's causing compliance failures, so if it needs to be there, please add:
> 
> ```
> /* Intentionally left empty */
> ```
> 
> to this file.

I don't think we need this. I believe it was added by mistake.

Reply:
> @lmaciejonczyk this is causing cyclic Kconfig dependencies.
> I disabled it here: [28e34ee](https://github.com/nrfconnect/sdk-nrfxlib/commit/28e34ee7f79a3cc66b2ff10daa7b4ff12df61f6f)
> 
> Could you take a second look at what is you real need and intentions, feel free to ping me directly:
> 

My intention was to enable FLASH module during activation of PSA_EITS_BACKEND_ZEPHYR since it was using it. If it's already active at present dependency chain I'm fine to remove the dependency.

Reply:
All has been resolved